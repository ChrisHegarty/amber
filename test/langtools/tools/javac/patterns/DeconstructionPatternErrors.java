/**
 * @test /nodynamiccopyright/
 * @summary Verify error reports for erroneous deconstruction patterns are sensible
 * @compile/fail/ref=DeconstructionPatternErrors.out --enable-preview -source ${jdk.version} -XDrawDiagnostics DeconstructionPatternErrors.java
 */

public class DeconstructionPatternErrors {

    public static void main(String... args) throws Throwable {
        Object p;
        p = new P(42);
        if (p instanceof P(String s));
        p = new P2(() -> {}, () -> {});
        if (p instanceof P2(String s, Runnable r)) {
            System.err.println("s=" + s);
            System.err.println("r=" + r);
        }
    }

    public record P(int i) {
    }

    public record P2(Runnable r1, Runnable r2) {}
}
