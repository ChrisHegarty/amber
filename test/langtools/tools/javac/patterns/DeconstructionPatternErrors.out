DeconstructionPatternErrors.java:12:28: compiler.err.deconstruction.pattern.no.var
DeconstructionPatternErrors.java:14:29: compiler.err.deconstruction.pattern.no.var
DeconstructionPatternErrors.java:14:39: compiler.err.deconstruction.pattern.no.var
- compiler.note.preview.filename: DeconstructionPatternErrors.java
- compiler.note.preview.recompile
3 errors
